Cargo.toml

```diff
@@ -16,4 +16,10 @@ keywords = ["data-structures"]
 readme = "README.md"
 
 [features]
-nightly = []
+nightly = ["serde"]
+
+[dependencies]
+serde = { optional = true, version = "^0.7" }
+
+[dev-dependencies]
+serde_json = "^0.7"
```

Comment:
Not in my experience. For running `cargo test` against linked-hash-map yes, which makes sense to me. Otherwise you would not know whether you broke the serde code. But for running `cargo test` against other crates that depend on linked-hash-map, it will not download and compile serde unless you are using the serde feature.